From ff0347a081e87816922ddf9fa4aa9e3df9bb7fe8 Mon Sep 17 00:00:00 2001
From: Rashed Abdel-Tawab <rashed@linux.com>
Date: Sat, 3 Mar 2018 10:26:14 -0800
Subject: [PATCH 02/22] build: Add ability for device to specify additional
 targets

Some devices (cough berkeley) need some additional targets like say
an image with just the kernel binary in it. Allow devices to specify
an "extension" to the main Makefile in their device trees.

Why do this whenwe have BOARD_CUSTOM_BOOTIMG_MK you might ask? Using
BOARD_CUSTOM_BOOTIMG_MK means you would have to duplicate the
bootimage and ramdiskimage generation code even if you're not changing
it, which is nasty. So add this and avoid any code duplication.

Change-Id: Ic34181518838c31716ea6a32e18545e7e93c6964
---
 core/Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/Makefile b/core/Makefile
index b58fd7d8f..c5ed03867 100644
--- a/core/Makefile
+++ b/core/Makefile
@@ -1409,6 +1409,10 @@ ifdef BOARD_CUSTOM_BOOTIMG_MK
 include $(BOARD_CUSTOM_BOOTIMG_MK)
 endif
 
+ifdef BOARD_CUSTOM_BUILD_MAKEFILE
+include $(BOARD_CUSTOM_BUILD_MAKEFILE)
+endif
+
 
 # -----------------------------------------------------------------
 # system image
-- 
2.17.1

